inc/query/query.php

```diff
@@ -152,4 +141,49 @@ public function handle_query_sibling( &$clauses, $id_column ) {
 
 		$clauses['where'] .= empty( $clauses['where'] ) ? $where : " AND $where";
 	}
+
+	/**
+	 * Modify query join & where statement for multi-relationship.
+	 *
+	 * @param string $clauses   Query clauses.
+	 * @param array  $args   $WP_query args object.
+	 */
+	public function handle_multiple_relationships( &$clauses, $id_column ) {
+		global $wpdb;
+		$relation = $this->args['relation'];
+		unset( $this->args['relation'] );
+		$relationships = $this->args;
+		$objects       = array();
+
+		foreach ( $relationships as $relationship ) {
+			$type          = $relationship['id'];
+			$source        = $relationship['direction'];
+			$items         = implode( ',', $relationship['items'] );
+			$items         = 1 == $relationship['reciprocal'] ? "(`from` IN ($items) OR `to` IN ($items))" : "`$source` IN ($items)";
```

Comment:
Chỗ này check `$relationship['reciprocal']` có được set hay ko? A nghĩ check `empty( $relationship['reciprocal'] )` là ổn (cần đảo ngược đoạn sau).

Reply:
Đoạn này a nghĩ với reciprocal thì tách ra làm 2 query: 1 query lấy cột `from`, 1 query lấy cột `to` cho chắc. Làm kiểu này tuy code gọn nhưng ko đảm bảo lấy được đúng em ạ. Nhất là đoạn bên dưới, em ko thể check theo kiểu này được (cho reciprocal):



```
$object_ids[] = 'from' === $source ? $result->to : $result->from;
```